src/api/interappbus/channel/index.ts

```diff
@@ -95,6 +95,14 @@ export class Channel extends EmitterBase<ChannelEvents> {
         const channel = new ChannelProvider(providerIdentity, this.wire.sendAction.bind(this.wire));
         const key = providerIdentity.channelId;
         this.channelMap.set(key, channel);
+        this.on('client-disconnected', (eventPayload: ProviderIdentity) => {
```

Comment:
Is this raised on window close from the core as well ?

Reply:
it uses the subscription manager so should work on window or frame close and external connection close

Reply:
i.e. client-disconnected is raised on those other events